funcs_utils.php

```diff
@@ -287,15 +287,18 @@ function current_branch_cms_major(
         return MetaData::HIGHEST_STABLE_CMS_MAJOR;
     }
 
+    $contents = '';
     if ($composerJson) {
         $contents = $composerJson;
     } elseif (check_file_exists('composer.json')) {
         $contents = read_file('composer.json');
     }
     $composerJson = json_decode($contents);
     if (is_null($composerJson)) {
-        $lastError = json_last_error();
-        error("Could not parse from composer.json - last error was $lastError");
+        if (!check_file_exists('composer.json') && !is_misc()) {
+            $lastError = json_last_error();
+            error("Could not parse from composer.json - last error was $lastError");
+        }
```

Comment:
Sorry, I had the suggestion slightly wrong in https://github.com/silverstripe/module-standardiser/pull/51#discussion_r1606185451.
We only want to emit the error if the file exists _or_ this isn't a misc repo. Assuming I understand the intention correctly.
```suggestion
        if (check_file_exists('composer.json') || !is_misc()) {
            $lastError = json_last_error();
            error("Could not parse from composer.json - last error was $lastError");
        }
```

Reply:
`if (!check_file_exists('composer.json') && !is_misc()) {` is correct

Intention is:

"If there is no composer.json, then error, unless is_misc()"

Reply:
If there is a file and we get an error, we also want the error.

So, we want the error if:
1. there is a file
2. there is no file, but this is not a misc repo

We do not want the error if:
1. This is a misc repo explicitly without the file.

Reply:
Currently this doesn't give us an error if there's a problem with decoding the JSON when a file does exist.

Reply:
Updated